src/rgw/sts-assume-role.cc

```diff
@@ -32,7 +32,7 @@ void Credentials::dump(Formatter *f) const
   encode_json("SessionToken", sessionToken , f);
 }
 
-int Credentials::generateCredentials(CephContext* cct)
+int Credentials::generateCredentials(CephContext* cct, const uint64_t& duration)
```

Comment:
should duration be a std::chrono unit?

Reply:
Could be, but we can always feed it to the std::chrono::seconds() constructor to make it one if we're going to use it with time_points.